backend/infrahub/workers/dependencies.py

```diff
@@ -0,0 +1,136 @@
+from fast_depends import Depends, inject
+from infrahub_sdk.client import InfrahubClient
+from infrahub_sdk.config import Config
+
+from infrahub import config
+from infrahub.components import ComponentType
+from infrahub.database import InfrahubDatabase, get_db
+from infrahub.services import InfrahubServices
+from infrahub.services.adapters.cache import InfrahubCache
+from infrahub.services.adapters.event import InfrahubEventService
+from infrahub.services.adapters.http import InfrahubHTTP
+from infrahub.services.adapters.http.httpx import HttpxAdapter
+from infrahub.services.adapters.message_bus import InfrahubMessageBus
+from infrahub.services.adapters.workflow import InfrahubWorkflow
+from infrahub.services.adapters.workflow.local import WorkflowLocalExecution
+from infrahub.services.adapters.workflow.worker import WorkflowWorkerExecution
+
+client: InfrahubClient | None = None
+database: InfrahubDatabase | None = None
+cache: InfrahubCache | None = None
+message_bus: InfrahubMessageBus | None = None
+event_service: InfrahubEventService | None = None
+workflow: InfrahubWorkflow | None = None
+http: InfrahubHTTP | None = None
+services: InfrahubServices | None = None
+
+
+def build_client() -> InfrahubClient:
+    global client
+    if client is None:
+        client = InfrahubClient(config=Config(address=config.SETTINGS.main.internal_address, retry_on_failure=True))
+    return client
+
+
+@inject
+def get_client(client: InfrahubClient = Depends(build_client)) -> InfrahubClient:  # noqa: B008
+    return client
```

Comment:
Thinking about this one, I think we might want it not to be a singleton but instead having a fresh instance of the client each time a flow needs one and pass it down the line of tasks that also need it.

Reply:
There are pros and cons to this, if we reuse the existing client it could be prepopulated with a schema so that we don't need to query the API, this can of course also be a bad thing if we don't actually want the schema to exist (perhaps it needs to be updated etc).

Another scenario is if our client will use the store, in that case perhaps we don't want to use the same client. We have the .clone method on the client and we could use other ways to determine what type of client we get depending on how we inject it. I think we can come back to that part once this is in place though.